app/src/main/java/space/taran/arknavigator/mvp/presenter/GalleryPresenter.kt

```diff
@@ -150,7 +164,39 @@ class GalleryPresenter(
         viewState.displayPreviewTags(resource.id, tags)
     }
 
-    private fun onPreviewsItemClick(itemView: PreviewItemView) {
+    private fun onRemovedResourceDetected() = presenterScope.launch {
+        viewState.setProgressVisibility(true, "Indexing")
+
+        index.reindex()
+        // update current tags storage
+        tagsStorageRepo.provide(rootAndFav)
```

Comment:
Am I right this line triggers garbage collection in tag storage?

Reply:
Yes, also registers new resources. I think we need a clearer way to do this. Perhaps the tags storage should have a reference to the index.

Reply:
I think index and storage will be merged into one bigger entity ("database"?) in upcoming library.